[prometheus-adapter] remove support for deprecated APIs (#5809)
prometheus-community/helm-charts: charts/prometheus-adapter/README.md

_See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._

### To 5.0.0

> This version removes support for the deprecated Kubernetes API `policy/v1` for PDBs. and support for Pod Security Policy.

Reviewer: We should not that Kubernetes >= 1.21 is now required.

> This version removes support for the deprecated Kubernetes API `policy/v1` for PDBs. and support for Pod Security Policy.

Reviewer: ```suggestion
This version removes support for the deprecated Kubernetes API `policy/v1beta1` for PodDisruptionBudgets. It also removes the deprecated `PodSecurityPolicy` resource.
```
